app/helpers/vertical_nav_helper.rb

```diff
@@ -257,7 +258,8 @@ def service_integration_items
   # Backend APIs
   def backend_api_nav_sections
     sections = []
-    return sections unless @backend_api
+    return sections unless @backend_api.persisted?
```

Comment:
Could you create one then? Or do you think it is unecessary

This actually really solved something, we should at least add a test to make it clear.

Reply:
@nidhi-soni1104 
Now the Vertical menu is not crashing, but you are returning an empty array, then there is no menu at all:

![01](https://user-images.githubusercontent.com/13486237/140728919-4ecd6eb3-44af-4f39-ab4b-1faa050e6350.png)

Reply:
The empty vertical menu might have something to do with the `activate_menu` method?. For actoin `#new` it calls `:backend_apis` but for `#show` it is https://github.com/3scale/porta/blob/ff78ef29249d2081983efa909b573a33ef3d6b50/app/controllers/provider/admin/backend_apis_controller.rb#L9

Reply:
Yes, I think if you add `activate_menu :backend_apis` in the create method, it will fix it
https://github.com/3scale/porta/blob/ff78ef29249d2081983efa909b573a33ef3d6b50/app/controllers/provider/admin/backend_apis_controller.rb#L22

Reply:
@josemigallas @damianpm Let me check this.

Reply:
I checked adding this to create action and it removed the vertical menu when create fails and I think it looks good. 

I am adding a new commit adding `activate_menu :backend_api` in create method,